rpc/network.go

```diff
@@ -0,0 +1,29 @@
+package rpc
+
+import (
+	"context"
+
+	capnp "capnproto.org/go/capnp/v3"
+)
+
+type PeerId struct {
```

Comment:
Is it reasonable to remove this and make the interfaces accept a type parameter instead?  It would be quite elegant to be able to reuse libp2p's `peer.ID` type, and I think it would also make sense for the type system to distinguish between, say, `Network[peer.ID]` and `Network[uuid.UUID]`.